Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/validation/ui/validationresult/open/DccValidationOpenFragment.kt

```diff
@@ -1,5 +1,61 @@
 package de.rki.coronawarnapp.covidcertificate.validation.ui.validationresult.open
 
+import android.os.Bundle
+import android.view.View
 import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.navArgs
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.covidcertificate.validation.ui.validationresult.common.ValidationResultAdapter
+import de.rki.coronawarnapp.covidcertificate.validation.ui.validationresult.common.setHeaderForState
+import de.rki.coronawarnapp.databinding.CovidCertificateValidationResultFragmentsBinding
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.lists.decorations.RecylerViewPaddingDecorator
+import de.rki.coronawarnapp.util.lists.diffutil.update
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.popBackStack
+import de.rki.coronawarnapp.util.ui.viewBinding
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import javax.inject.Inject
 
-class DccValidationOpenFragment : Fragment()
+class DccValidationOpenFragment : Fragment(R.layout.covid_certificate_validation_result_fragments), AutoInject {
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val binding by viewBinding<CovidCertificateValidationResultFragmentsBinding>()
+    private val args by navArgs<DccValidationOpenFragmentArgs>()
+    private val viewModel: DccValidationOpenViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as DccValidationOpenViewModel.Factory
+            factory.create(
+                validation = args.validation
+            )
+        }
+    )
+
+    @Inject lateinit var validationResultAdapter: ValidationResultAdapter
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        binding.apply {
+            setHeaderForState(args.validation.state)
+            list.apply {
+                adapter = validationResultAdapter
+                addItemDecoration(
+                    RecylerViewPaddingDecorator(
+                        topPadding = R.dimen.spacing_small,
+                        leftPadding = R.dimen.spacing_small,
+                        rightPadding = R.dimen.spacing_small,
+                    )
+                )
+            }
+
+            toolbar.setNavigationOnClickListener { popBackStack() }
+        }
+
+        viewModel.listItems.observe2(this) {
+            validationResultAdapter.update(it)
+        }
+
+        super.onViewCreated(view, savedInstanceState)
```

Comment:
I think it is not necessary to call super method here , it is empty actually , right?

Reply:
I'd prefer if we keep calling super on the standard lifecycle methods from Fragments and Activities, while they may be empty now, if we extend fragments for something in the future, or AndroidX libraries change Fragment implementation details, we'd end up with weird bugs.

Reply:
I don't mind as I said it is actually empty, but we don't do this btw anywhere else in the App  , does this mean that we will have bugs everywhere ?. I think the only reason why it is used here is `apply`

if we will use super methods , then I think it should be moved as first line in the method body , not a the end ,right?

Reply:
yes, it makes sense, updated